services/server/src/handlers/user.ts

```diff
@@ -653,6 +653,67 @@ async function findStudentStats(
   res.status(200).json(response);
 }
 
+async function findMyStats(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const user = req.user;
+  const allowed = isStudent(user);
+  if (!allowed) return next(forbidden());
+
+  const id = user.id;
+  const studentData = await users.findById(id);
+  if (!studentData) return next(notfound.student());
+
+  const nowdate = dayjs.utc().toISOString();
+
+  const tutorCount = await lessons.countCounterpartMembers({
+    user: id,
+    ratified: true,
+    canceled: true,
+  });
+
+  const completedLessonCount  = await lessons.countLessons({
+    users: [id],
+    ratified: true,
+    canceled: false,
+    before: nowdate
+  });
+
+  const totalMinutes = await lessons.sumDuration({
+    users: [id],
+    ratified: true,
+    canceled: true,
+    before: nowdate
+  });
+
+  const canceledMinutes = await lessons.sumDuration({
+    users: [id],
+    ratified: false,
+    canceled: true,
+    before: nowdate
+  });
+
+  const totalLearningTime = totalMinutes - canceledMinutes;
```

Comment:
No need to do this. The total learning time is simply can be queried like this. 

```suggestion
  const totalLearningTime = await lessons.sumDuration({
    users: [id],
    ratified: true,
    canceled: true,
    before: nowdate
  });
```